File: tests/06_phases_openmp.dg/common/cxx/success_task_05.cpp
/*
<testinfo>
test_generator=config/mercurium-omp

test_compile_fail_nanos4_instrument=yes
test_compile_faulty_nanos4_instrument=yes
test_compile_fail_nanos4_plain=yes
test_compile_faulty_nanos4_plain=yes
<</testinfo>
*/
#include <stdlib.h>

struct A
{
template <typename T>
    void foo(int n);
};

struct C
{
    int n;
    C() : n(0) { }
};
C e;

template <typename Q>
void A::foo(int n)
{
#pragma omp task shared(e) firstprivate(n)
    {
        e.n = n;
    }
}

int main(int argc, char *argv[])
{
    A a;

    a.foo<int>(4);

#pragma omp taskwait

    if (e.n != 4)
        abort();

    return 0;
}

File: tests/07_phases_ompss.dg/fortran/success_task_05.f90
! <testinfo>
! test_generator=config/mercurium-ompss
! </testinfo>

MODULE M
IMPLICIT NONE
TYPE MATRIX_T
    INTEGER :: I
END TYPE

CONTAINS
    SUBROUTINE MAT_VEC(MAT)
        IMPLICIT NONE
        TYPE (MATRIX_T) :: MAT

        !$OMP TASK SHARED(MAT)
            MAT % I = 1
        !$OMP END TASK
    END SUBROUTINE MAT_VEC

END MODULE M

PROGRAM P
    USE M
    IMPLICIT NONE

    TYPE(MATRIX_T) :: MAT
    MAT % I = 0
    CALL MAT_VEC(MAT)
    !$OMP TASKWAIT
END PROGRAM P
